doc/classes/vector.adoc

```diff
@@ -0,0 +1,256 @@
+* `v`, `axis`: `vector`
+* `angle`: `number`
+
+Returns a new `vector` which is equal to `v` rotated around `axis` by `angle` radians.
+
+=== `vector.rotate(v, rot) -> vector`
+* `v`, `rot`: `vector`
+
+Returns a new `vector` which is equal to `v` rotated by `rot`.
+
+The way that the components of `rot` map is as follows:
+
+* `rot.x` is pitch
+* `rot.y` is yaw
+* `rot.z` is roll
+
+=== `vector.dir_to_rotation(forward, up) -> vector`
+* `forward`: `vector`
+* `up`: `vector` or `nil`
+
+Returns a new rotational `vector` (the same kind as `rot` in `vector.rotate()`) equal to the rotation from `up` to `forward`.
+
+If `up` is `nil` then the returned rotational `vector` assumed that `y = 1` is up.
+
+Both `up` and `forward` are normalized by the function before calculations are made with them, so a call with or without normalization by the caller will be the same.
+
+== Metatable Functions
+
+Metatable-enhanced `vector` values have some convenience features to help make vector math more readable.
+They can have be used with normal math operations rather than needing to call the equivalent namespaced function.
+They also can be indexed either with named keys (`v.x` and `v["x"]`) or they can be indexed with numeric keys (`v[1]` being `v.x``, `v[2]` being `v.y`, and `v[3]` being `v.z`).
+
+.Using Metatable `vectors`
+====
+
+[source, lua]
+----
+local v1 = vector.new(1, 2, 3)
+local v2 = vector.new(4, 5, 6)
+local v3
+
+-- This:
+v3 = vector.add(v1, v2)
+-- Is equivalent to:
+v3 = v1 + v2
+
+-- You can also use do more lengthy calculations:
+v3 = ((v1 + v2) / 2) * 3
+-- The equivalent using the namespaced functions would be rather unwieldy:
+v3 = vector.multiply(vector.divide(vector.add(v1, v2), 2), 3)
+----
+
+====
+
+=== `metatable.__eq(a, b) -> bool`
+* `a`, `b`: `vector`
+
+Literally `vector.equals(a, b)`.
+
+=== `metatable.__add(a, b) -> vector`
+* `a`, `b`: `vector`
+
+Returns a new `vector` where each component of `a` is added to each respective component of `b`.
+
+NOTE: Unlike `vector.add()` this does not support adding `number` values.
+
+=== `metatable.__sub(a, b) -> vector`
+* `a`, `b`: `vector`
+
+Returns a new `vector` where each component of `a` is subtracted from each respective component of `b`.
+
+NOTE: Unlike `vector.subtract()` this does not support subtracting `number` values.
+
+=== `metatable.__mul(a, b) -> vector`
+* `a`: `vector` or `number`
+* `b`: `number` or `vector`
+
+Returns a new `vector` where each component of `a` is multiplied by each respective component of `b`. Because of the way metatables work, either argument can be a `number` or a `vector` without any practical difference.
+
+=== `metatable.__div(a, b) -> vector`
+* `a`: `vector`
+* `b`: `number`
+
+Returns a new `vector` where each component of `a` is divided by `b`.
+
+WARNING: This function assumes `b` is a number, and will probably return confusing errors about "accessing a `nil` value" if it is not.
```

Comment:
All metatable functions will return confusing errors as they don't contain assertions (presumably for performance reasons)

Reply:
It's good to note if it's likely to be very unintuitive though.